doc/tutorial/basic/tutorial.rst

```diff
@@ -250,3 +250,102 @@ Note that pickle has some security and maintainability issues. Please refer to
 section :ref:`model_persistence` for more detailed information about model
 persistence with scikit-learn.
 
+
+Conventions
+-----------
+
+scikit-learn estimators follow certain rules to make their behavior more
+predictive.
+
+
+Type casting
+~~~~~~~~~~~~
+
+Unless otherwise specified, input will be cast to ``float64``::
+
+  >>> import numpy as np
+  >>> from sklearn import random_projection
+
+  >>> rng = np.random.RandomState(0)
+  >>> X = rng.rand(10, 2000)
+  >>> X = np.array(X, dtype='float32')
+  >>> X.dtype
+  dtype('float32')
+
+  >>> transformer = random_projection.GaussianRandomProjection()
+  >>> X_new = transformer.fit_transform(X)
+  >>> X_new.dtype
+  dtype('float64')
+
+In this example, ``X`` is ``float32``, which is cast to ``float64`` by
+``fit_transform(X)``.
+
+Regression targets are cast to ``float64``, classification targets are
+maintained::
+    >>> from sklearn import datasets
+    >>> from sklearn.svm import SVC
+
+    >>> iris = datasets.load_iris()
+    >>> clf = SVC()
+    >>> clf.fit(iris.data, iris.target)
+    SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0, degree=3, gamma=0.0,
+      kernel='rbf', max_iter=-1, probability=False, random_state=None,
+      shrinking=True, tol=0.001, verbose=False)
+
+    >>> clf.predict(iris.data[:3])
+    array([0, 0, 0])
+
+    >>> clf.fit(iris.data, iris.target_names[iris.target])
+    SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0, degree=3, gamma=0.0,
+      kernel='rbf', max_iter=-1, probability=False, random_state=None,
+      shrinking=True, tol=0.001, verbose=False)
+
+    >>> clf.predict(iris.data[:3])  # doctest: +NORMALIZE_WHITESPACE
+    array(['setosa', 'setosa', 'setosa'], dtype='<U10')
```

Comment:
This is causing the test to fail on old versions of Python. To avoid that we could do:

```
>>> predicted = clf.predict(iris.data[:3])
>>> list(predicted)
['setosa', 'setosa', 'setosa']
```